packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -38,53 +38,36 @@ abstract class UnpackMacOS extends Target {
   ];
 
   @override
-  List<Source> get outputs => const <Source>[];
+  List<Source> get outputs => const <Source>[
+    Source.pattern('{OUTPUT_DIR}/FlutterMacOS.framework/FlutterMacOS'),
```

Comment:
Either it's already been copied to `build/macos/Build/Products/Debug` or it hasn't, so we don't need `unpack_macos.d` anymore.   The subclasses include `Source.artifact(Artifact.flutterMacOSFramework, mode: x)` which will cause re-copies on flutter upgrades.